packages/shared/src/server/queues.ts

```diff
@@ -20,6 +21,29 @@ export const LegacyIngestionEvent = z.object({
   }),
 });
 
+export const LegacyIngestionEventMeta = z.object({
+  useS3EventStore: z.literal(true),
+  data: z.array(
+    z.object({
+      type: z.nativeEnum(eventTypes),
+      eventBodyId: z.string(),
+      eventId: z.string(),
+    }),
+  ),
+  authCheck: z.object({
+    validKey: z.literal(true),
+    scope: z.object({
+      projectId: z.string(),
+      accessLevel: z.enum(["all", "scores"]),
+    }),
+  }),
+});
+
+export const LegacyIngestionEvent = z.discriminatedUnion("useS3EventStore", [
```

Comment:
thx for the backwards compatibility

Reply:
@maxdeichmann I've also verified that an event without the `useS3EventStore` property is processed successfully on new workers, i.e. we're also good during the deployment _and_ have this as a fallback.